sklearn/feature_selection/from_model.py

```diff
@@ -111,3 +135,98 @@ def transform(self, X, threshold=None):
             return X[:, mask]
         else:
             raise ValueError("Invalid threshold: all features are discarded.")
+
+
+class SelectFromModel(BaseEstimator, SelectorMixin):
+    """Meta-transformer for selecting features based on importance weights.
+
+    Parameters
+    ----------
+    estimator : object
+        The base estimator from which the transformer is built.
+        This can be both a fitted or a non-fitted estimator.
+        If it a fitted estimator, then ``transform`` can be called directly,
+        otherwise train the model using ``fit`` and then ``transform`` to do
+        feature selection.
+
+    threshold : string, float, optional
+        The threshold value to use for feature selection. Features whose
+        importance is greater or equal are kept while the others are
+        discarded. If "median" (resp. "mean"), then the ``threshold`` value is
+        the median (resp. the mean) of the feature importances. A scaling
+        factor (e.g., "1.25*mean") may also be used. If None and if
+        available, the object attribute ``threshold`` is used. Otherwise,
+        "mean" is used by default.
+
+    Attributes
+    ----------
+    `estimator_`: an estimator
+        The base estimator from which the transformer is built.
+        This is stored only when a non-fitted estimator is passed to the
+        ``SelectFromModel``.
+
+    `threshold_`: float
+        The threshold value used for feature selection.
+    """
+    def __init__(self, estimator, threshold=None):
+        self.estimator = estimator
+        self.threshold = threshold
+
+    def _get_support_mask(self):
+        # SelectFromModel can directly call on transform.
```

Comment:
This was suggested by @glouppe here (https://github.com/scikit-learn/scikit-learn/pull/4242#discussion-diff-40066983). I do not really have an opinion on this. But I do agree that it violates the `fit-predict` API and it is almost trivial to mask the feature importances oneself.

But I suppose we can leave this debatable feature open as an issue and remove it for now?